api/v1beta1/cryostat_types.go

```diff
@@ -118,7 +118,7 @@ type CryostatStatus struct {
 	ApplicationURL string `json:"applicationUrl"`
 	// Grafana secret
 	// +operator-sdk:csv:customresourcedefinitions:type=status
-	GrafanaSecret *corev1.Secret `json:"grafanaSecret"`
+	GrafanaSecret string `json:"grafanaSecret"`
```

Comment:
Could you add the `// +optional` marker for this field, and also append `omitempty` to the JSON tag? Optional tells the Kubernetes API server that this field is not required, such as before the secret is created. `omitempty` is similar, but functions on the JSON level. If the field is missing, it won't be transmitted in the JSON body.

There's one more marker I'd like to add here. If you look at other fields, you'll see an `xDescriptors` marker. This controls some generated UI in the OpenShift Console. We can tell the Console that this field refers to the name of a Kubernetes Secret. When it gets rendered in the OpenShift Console, it should create a link to the named Secret that users can click on. The example here conveniently shows how this works for Secrets, but also works for other kinds of Kubernetes objects: https://github.com/openshift/console/blob/master/frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md#3-k8sResourcePrefix